pandas/core/groupby/generic.py

```diff
@@ -1266,15 +1266,24 @@ def first_not_none(values):
                                 axis=self.axis,
                             ).unstack()
                             result.columns = index
-                    else:
+                    elif isinstance(v, ABCSeries):
                         stacked_values = np.vstack([np.asarray(v) for v in values])
                         result = DataFrame(
                             stacked_values.T, index=v.index, columns=key_index
                         )
+                    else:
+                        # GH#1738: values is list of arrays of unequal lengths
```

Comment:
a better way to do this would be to make this if/elif into a function, then just return on the if/elif; that way you can avoid suplicating 1279 and 1287

Reply:
generally agree, but I _think_ the version on 1287 doesnt need to be there at all (not reached in tests).  Just not removing it until im sure

Reply:
ok, then merge this and followup?

Reply:
yes please.  got a couple of local branches about ready to go